File: dotnet/private/tests/stamping/StampingTest.fs
open System
open System.Reflection

[<EntryPoint>]
let main _ =
    let expected = "1.2.3-beta.4+abcdef"
    let attribute = Assembly.GetExecutingAssembly().GetCustomAttribute<AssemblyInformationalVersionAttribute>()
    let actual = if isNull attribute then null else attribute.InformationalVersion

    if actual = expected then
        0
    else
        eprintfn "Expected %s, got %s" expected (if isNull actual then "<null>" else actual)
        1

File: dotnet/private/tests/stamping/StampingTest.cs
using System;
using System.Reflection;

public static class Program
{
    public static int Main()
    {
        const string expected = "1.2.3-beta.4+abcdef";
        var actual = Assembly.GetExecutingAssembly()
            .GetCustomAttribute<AssemblyInformationalVersionAttribute>()
            ?.InformationalVersion;

        if (actual == expected)
        {
            return 0;
        }

        Console.Error.WriteLine($"Expected {expected}, got {actual ?? "<null>"}");
        return 1;
    }
}
